### PR TITLE
Add AELO release information

### PR DESCRIPTION
**Pull request by ptormene**
The idea is to read the release information from a changelog file and display to the user portions of it.

### CODE REVIEW

**Comment by micheles** [comment]
Here is a proposal for the changelog file, to be discussed:
```
[Release 2.0.0]
Date: 2024-12-31
Requires: engine-3.22
Notes: Final version
Private: see file XXX

[_Release 1.1.0]
Requires: engine-3.21
Date: 2024-09-31
Notes:
  Preliminary version only available on the beta machine.
  It has the following features:
  1. extended to 10 IMTs
  2. implemented ASCE17
  3. ...
Private: see file XXX

[Release 1.0.0]
Date: 2024-06-31
Requires: engine-3.20
Notes: Initial version
Private: see file XXX
```

**Comment by ptormene** [comment]
I modified the implementation as suggested by @micheles.
If APPLICATION_MODE='AELO', a `aelo_changelog.ini` file must be placed into the `mosaic_dir`.
Follows an example of changelog file and the corresponding rendering in the web application:

```ini
[Release 2.0.0]
Date: 2024-12-31
Requires: engine-3.22
Notes: Final version
Private: see file XXX

[Release 1.1.0]
Date: 2024-09-30
Requires: engine-3.21
Notes:
  Preliminary version only available on the beta machine.
  It has the following features:
  1. extended to 10 IMTs
  2. implemented ASCE17
  3. ...
Private: see file XXX

[Release _1.0.1]
Date: 2024-07-10
Requires: engine-3.20
Notes:
  Version not to be displayed in the changelog web page
Private: see file XXX

[Release 1.0.0]
Date: 2024-06-31
Requires: engine-3.20
Notes: Initial version
Private: see file XXX
```

image

**Comment by ptormene** [comment]
image

TODO: store the AELO version into the datastore and visualize it in the outputs page and in the exported files

**Comment by ptormene** [comment]
NOTE: the `aelo_version` is saved into the datastore only if the calculation runs through the webui with `APPLICATION_MODE == 'AELO'` in `local_settings.py` or via the command line using the environment variable `OQ_APPLICATION_MODE=AELO`.
If `aelo_version` is not in the datastore, the AELO outputs page will look as it was before this PR and the downloaded csv outputs will display the `engine_version` instead.

File: openquake/calculators/base.py [hunk 1]
```diff
@@ -67,6 +69,38 @@
...
+    c.read(changelog_path)
+    for sec in c.sections():
+        dic['AELO VERSION'].append(sec)
+        for k, v in c.items(sec):
+            dic[k].append(v)
+    df = pandas.DataFrame(dic)
+    df = df.drop(columns=['private'])
+    # NOTE: extracting version from a string like "AELO VERSION 1.0.0"
+    df['AELO VERSION'] = df['AELO VERSION'].apply(remove_heading_2_words)
+    df = df[~df['AELO VERSION'].str.startswith('_')]
```
Comment (micheles) [inline openquake/calculators/base.py:90]: I would remove this feature, i.e. change _1.1.0 -> 1.1.0

Comment (ptormene) [reply openquake/calculators/base.py]: Done

File: openquake/server/views.py [hunk 2]
```diff
@@ -1359,6 +1360,10 @@ def web_engine_get_outputs_aelo(request, calc_id, **kwargs):
         lon, lat = ds['oqparam'].sites[0][:2]  # e.g. [[-61.071, 14.686, 0.0]]
         vs30 = ds['oqparam'].override_vs30  # e.g. 760.0
         site_name = ds['oqparam'].description[9:]  # e.g. 'AELO for CCA'->'CCA'
+        try:
+            calc_aelo_version = ds.get_attr('/', 'aelo_version')
+        except KeyError:
+            calc_aelo_version = None
```
Comment (ptormene) [inline openquake/server/views.py:1366]: For old calculations we can assign version 1.0.0

**Comment by ptormene** [comment]
I simplified a bit the format of the aelo_changelog.ini file:

```ini
[2.1.0]
date: 2024-12-31
engine_version: engine-3.22
notes: Changed presentation
private: see file XXX

[2.0.0]
date: 2024-09-30
engine_version: engine-3.21
notes:
  Preliminary version only available on the beta machine.
  It has the following features:
  1. extended to 10 IMTs
  2. implemented ASCE17
  3. ...
private: see file XXX

[1.0.0]
date: 2024-06-31
engine_version: engine-3.20
notes: Initial version
private: see file XXX
```

If a calculation has not `aelo_version` stored in the datastore, version `1.0.0` is automatically assigned. This happens for the existing calculations in the currently running installations or when a calculation runs via the command line without the OQ_APPLICATION_MODE=AELO environment variable.

File: openquake/server/views.py (hunk 2 above)
Comment (ptormene) [reply openquake/server/views.py]: Done

**Review by micheles — APPROVED** [review]

**Pull request merged**